apps/dashboard/src/components/workflow-row.tsx

```diff
@@ -101,36 +126,57 @@ export const WorkflowRow = ({ workflow }: WorkflowRowProps) => {
       </TooltipProvider>
 
       <TableCell className="w-1">
-        <DropdownMenu>
+        <ConfirmationModal
+          open={isDeleteModalOpen}
+          onOpenChange={setIsDeleteModalOpen}
+          onConfirm={onDeleteStep}
+          title="Are you sure?"
+          description={`You're about to delete the ${workflow.name}, this action cannot be undone.`}
+          confirmButtonText="Delete"
+        />
```

Comment:
I'm not sure about the modal approach here as it closes right away after clicking on the delete btn. On slow network connections, the users might still see the workflow and interact with it, which can cause problems (especially since we have two requests: delete and refetch).  
So, I think that we should keep the modal open until the delete request is completed.

Reply:
Yeah fair point, will update the behaviour